src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -824,10 +831,13 @@ public int ExecuteOnThread(IDebugThread2 pThread)
         // that is, not all threads should be required to be stopped before this method returns. The implementation of this method may be 
         // as simple as calling the IDebugProgram2::CauseBreak method on this program.
         //
-        // The sample engine only supports debugging native applications and therefore only has one program per-process
         public int Stop()
         {
-            throw new NotImplementedException();
+            DebuggedProcess.WorkerThread.RunOperation(async () =>
+            {
+                await _debuggedProcess.CmdBreak();
```

Comment:
There is a known bug in during attach that CmdBreak doesn't currently wait until the signal is actually received by the debuggee (it returns as soon as the signal is sent but not before it is received). This await will therefore return too early if the original debuggee was attached to rather than launched. Is child process debugging support for attach on Linux?